File: 卡码/代码随想录/C++/哈希/05-四数相加/LeetCode_0454.cpp
//454. 四数相加 II
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
  int fourSumCount(vector<int>& nums1, vector<int>& nums2, vector<int>& nums3, vector<int>& nums4) {
		unordered_map<int,int> m;
		int sum=0;
		for(int num1:nums1)
			for(int num2:nums2)
				m[num1+num2]++;
		for(int num3:nums3)
			for(int num4:nums4){
				int target=-(num3+num4);
				sum+=m[target];
			}
		return sum;
  }
};

File: 卡码/代码随想录/Python/哈希/05-四数相加/LeetCode_0454.py
#454. 四数相加 II
from typing import *

class Solution:
	def fourSumCount(self, nums1: List[int], nums2: List[int], nums3: List[int], nums4: List[int]) -> int:
		map=dict()
		sum=0
		for num1 in nums1:
			for num2 in nums2:
				target=num1+num2
				map[target]=map.get(target,0)+1
		for num3 in nums3:
			for num4 in nums4:
				target=-(num3+num4)
				sum+=map.get(target,0)
		return sum